lib/pages/bucket_page/bucket_page.dart

```diff
@@ -66,81 +95,47 @@ class BucketPage extends StatelessWidget {
           List<Object>? objects = state.items.objects;
           List<ListTile>? directoriesWidgets = [];
 
-          directories.forEach(
-            (directory) {
-              List<String> splittedPrefix = directory.split('/');
-              String? directoryName = splittedPrefix.length > 2
-                  ? splittedPrefix[splittedPrefix.length - 2]
-                  : splittedPrefix[0];
-              List<String> newPath = [...state.path];
-              newPath.add(directoryName);
-
-              if (directoryName.contains(state.filter)) {
-                directoriesWidgets.add(ListTile(
-                  leading: const Icon(FluentIcons.folder_horizontal),
-                  title: Text(directoryName.replaceAll('/', '')),
-                  onPressed: () {
-                    context
-                        .read<BucketPageBloc>()
-                        .add(DirectoryAdded(path: newPath));
-                    context
-                        .read<BucketPageBloc>()
-                        .add(ObjectsRequested(prefix: newPath.join('/')));
-                  },
-                ));
-              }
-            },
-          );
+          for (var directory in directories) {
+            List<String> splittedPrefix = directory.split('/');
+            String? directoryName = splittedPrefix.length > 2
```

Comment:
can we use `path.dirname` or any built-in `path` lib method for this?

Reply:
I haven't touched this code. I'm not ready to decide how to do it right